catalyst/core/state.py

```diff
@@ -1,192 +1,133 @@
-from typing import Dict, Optional  # isort:skip
+from typing import Dict, Optional, Union, TYPE_CHECKING  # isort:skip
 from collections import defaultdict, OrderedDict
 from pathlib import Path
 
 import numpy as np
 
+from catalyst import utils
 from catalyst.utils.tools.frozen_class import FrozenClass
-from catalyst.utils.tools.metric_manager import MetricManager, TimerManager
 from catalyst.utils.tools.typing import (
-    Criterion, Device, Model, Optimizer, Scheduler
+    Criterion, DataLoader, Device, Model, Optimizer, Scheduler
 )
 
+if TYPE_CHECKING:
+    from .callback import Callback  # noqa: F401
+
+STATE_MODEL = Union[Model, Dict[str, Model]]
```

Comment:
why caps and not `StateModel` or something?

Reply:
sounds legit